### ISSUE TITLE
'react/no-unused-prop-types' reports false positives when using deep destructuring.

### ISSUE THREAD

**Issue by IUnknown68** [yannickcr/eslint-plugin-react#2444]
When deep-destructuring component properties (in a functional component), 'react/no-unused-prop-types' is reported for the top level property:

```js
import {
  intlShape,
  injectIntl,
} from 'react-intl';

import messages from './some-component.intl';

function SomeComponent({ intl: { formatMessage } }) {
  return (
    <img title={ formatMessage(messages.imageTitle) } />
  );
}

SomeComponent.displayName = 'SomeComponent';

SomeComponent.propTypes = {
  intl: intlShape.isRequired,
};

SomeComponent.defaultProps = {};

export default injectIntl(SomeComponent);
```
reports "'intl' PropType is defined but prop is never used".

Possible workaround:
```js
function SomeComponent({ intl }) {
  const { formatMessage } = intl;
  return (
    <img title={ formatMessage(messages.imageTitle) } />
  );
}
```
I checked open issues, but didn't find any matching this exact case.

**Comment by golopot** [comment]
I am 97% sure this is a duplicate of #2424, and is fixed by #2428, but not yet released.

**Comment by IUnknown68** [comment]
Yup, looks like it - didn't see that one for some reason. Thanks and have a good day!
